src/transactions/FeeBumpTransactionFrame.cpp

```diff
@@ -156,9 +159,33 @@ FeeBumpTransactionFrame::processPostApply(Application& app,
                                           AbstractLedgerTxn& ltx,
                                           TransactionMetaFrame& meta)
 {
+    int64_t preRefundFeeCharged = mInnerTx->getResult().feeCharged;
+
     // We must forward the Fee-bump source so the refund is applied to the
     // correct account
     mInnerTx->processPostApply(app, ltx, meta, getFeeSourceID());
+
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    // The result codes and a feeCharged without the refund are set in
+    // updateResult in FeeBumpTransactionFrame::apply. At this point, feeCharged
+    // is set correctly on the inner transaction, so update the feeBump result.
+    if (protocolVersionStartsFrom(ltx.loadHeader().current().ledgerVersion,
+                                  ProtocolVersion::V_21) &&
+        isSoroban())
+    {
+        // First update feeCharged of the inner result on the feeBump using
+        // mInnerTx
+        auto& irp = mResult.result.innerResultPair();
+        auto& innerRes = irp.result;
+        innerRes.feeCharged = mInnerTx->getResult().feeCharged;
+
+        // Now set update feeCharged on the fee bump. We don't have access to
+        // the refund value here, but we can calculate it using the pre and post
+        // feeCharged of mInnerTx.
+        int64_t refund = preRefundFeeCharged - mInnerTx->getResult().feeCharged;
```

Comment:
This is a kinda hacky, but it does make sure that any adjustment done to the inner fee charged is also applied to the feeBump feeCharged. We could also just expose the refund from `mSorobanExtension` in `TransactionFrame` instead, which will be cleaner.

Reply:
Using `mSorobanExtension` isn't trivial though because there are some cases where the refund is skipped - account doesn't exist or liabilities are too high, so that will also need to be taken into account.

Reply:
yeah this looks pretty foot gunny:
* inner tx's `feeCharged` should be 0 in this case (consistent with classic)
* modifying inner tx from the outside like that is brittle

I would recommend refactoring code a bit (maybe add a new virtual function to deal with this)